core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -615,4 +630,45 @@ private static void updateTotal(
       }
     }
   }
+
+  private Long snapshotIdToRollbackToOnConflict(Long parentSnapshotId) {
+    Long newParentSnapshotId = parentSnapshotId;
+    if (PropertyUtil.propertyAsBoolean(
```

Comment:
can we move this property check outside this method? So we don't have such a long if block.